src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash-es';
+
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+function deleteCategory(allRelationships, category) {
+  delete allRelationships[category];
+  return allRelationships;
+}
+
+function deleteRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+) {
+  const targetRelationshipCategory = get(
+    allRelationships,
+    currentCategory,
+  );
+  const relationshipsWithoutTarget = targetRelationshipCategory.filter(
+    entry => entry !== relationship,
+  );
+  allRelationships[currentCategory] = relationshipsWithoutTarget;
+  return allRelationships;
```

Comment:
it's usually bad practice to return a mutated version of the same object in functions like this. it's one of the areas JS is weird and debugging can get really hard. best practice is something like 

```
return {
    ...allRelationships,
    [currentCategory]: relationshipsWithoutTarget
}
```

Reply:
I always get super confused about what needs deep copies and what does not.